### PR TITLE
Completed Challenge and Review Pages to Final Form

### PR DESCRIPTION
**Pull request by fru1tstand**
+ Added mock challenge question
+ Completed live syntax highlighting "within" textarea
+ Added review code commenting
+ Wired score metric assignment

### CODE REVIEW

**Comment by fru1tstand** [comment]
Finishes #29, and #31

File: src/main/resources/templates/challenge.html [hunk 1]
```diff
@@ -15,12 +15,185 @@
...
+			</ul>
+			<div class="block">
+				Any other non-listed assumptions must be handled appropriately.
+			</div>
+		</div>
+	</div>
+	<div class="right">
+		<div>
+			<textarea class="code-highlight" id="challenge-input">import java.lang.annotation.Annotation;
+import java.lang.reflect.AnnotatedElement;
```
Comment (jonathanpli) [inline src/main/resources/templates/challenge.html]: Not sure how to request a change for this, but when looking at the front end, if I try interacting with the code, a bunch of the code in the challenge shows up with red underline, even though the package imports look correct. Would you happen to know how to fix that?

**Review by jonathanpli — CHANGES_REQUESTED** [review]

File: src/main/resources/templates/challenge.html (hunk 1 above)
Comment (fru1tstand) [reply src/main/resources/templates/challenge.html]: Yeah, it's the browser's autocorrect. Good catch.

Comment (jonathanpli) [reply src/main/resources/templates/challenge.html]: Cool! Seemed to fix most of the problems. There are still some red underlining, but here's an interesting one -- 
screen shot 2017-05-25 at 1 58 56 am

So there's the red underline, but if I try deleting one character, this is what happens --
screen shot 2017-05-25 at 1 59 53 am

EDIT: It looks like the indentation is just incorrect for all the code--the caret doesn't match where the input displays

EDIT 2: This problem only occurs on smaller screens; it works fine on my larger monitor

**Review by jonathanpli — CHANGES_REQUESTED** [review]

File: src/main/resources/templates/challenge.html (hunk 1 above)
Comment (fru1tstand) [reply src/main/resources/templates/challenge.html]: I didn't push to the server, but can you highlight the part that's underlined and screenshot it again? The way it's set up is a <textarea> is laying over the \<pre>\<code> block and the styles between the two should match exactly. The textarea is providing the ability to edit, show a caret, and highlight (not color, think copy/paste type of highlight) text, and the pre/code block is there to provide the syntax highlighting. The underlining of what appears to be blank space may be caused by the two not aligning properly.

Comment (jonathanpli) [reply src/main/resources/templates/challenge.html]: Here's the text alignment problem--the red underline is gone now, interestingly enough.

screen shot 2017-05-25 at 2 37 22 am

Comment (fru1tstand) [reply src/main/resources/templates/challenge.html]: I pushed to the server which removed all underlining. I also fixed the text collapse on all major browsers except IE. IE will never stop softwrapping the textarea which is throwing the alignment off. For now, I don't think I want to invest any more time into the issue as there are more important things than getting IE compliance. Let me know if any more issues arise in Firefox, Chrome, or Safari.

Comment (jonathanpli) [reply src/main/resources/templates/challenge.html]: All of the underlining is gone (yay!), but the text alignment issue is still there for my smaller screen on Chrome, even after doing a hard cache refresh. Assuming we'll be using a larger monitor for the demo, and it doesn't affect the demo, though, I think it's fine to ignore the problem for now.

image

**Review by jonathanpli — APPROVED** [review]

**Pull request merged**